references/cli/addon.go

```diff
@@ -171,25 +185,42 @@ func NewAddonUpgradeCommand(c common.Args, ioStream cmdutil.IOStreams) *cobra.Co
 			if err != nil {
 				return err
 			}
-			_, err = pkgaddon.FetchAddonRelatedApp(context.Background(), k8sClient, name)
-			if err != nil {
-				return errors.Wrapf(err, "cannot fetch addon related addon %s", name)
-			}
 			addonArgs, err := parseToMap(args[1:])
 			if err != nil {
 				return err
 			}
-			err = enableAddon(ctx, k8sClient, config, name, addonArgs)
-			if err != nil {
-				return err
+			addonOrDir := args[0]
```

Comment:
what if the local directory has the same name as an addon in the registry? maybe we should add a bool flag to mark. How about `--dev` or `--local`

Reply:
Have considered this situation. But IMO this is fine. The main logic is  if local truly have this path will install it from local dir otherwise will install from remote registry. Refer to the implementation of helm https://github.com/helm/helm/blob/292368a64d6c9794b4573afa5a8dc6e940a21045/pkg/action/install.go#L668